scripts/linters/html_linter.py

```diff
@@ -0,0 +1,433 @@
+    """
+    def __init__(
+            self, files_to_lint, verbose_mode_enabled=False):
+        """Constructs a HTMLLintChecksManager object.
+
+        Args:
+            files_to_lint: list(str). A list of filepaths to lint.
+            verbose_mode_enabled: bool. True if verbose mode is enabled.
+        """
+        self.files_to_lint = files_to_lint
+        self.verbose_mode_enabled = verbose_mode_enabled
+
+    @property
+    def html_filepaths(self):
+        """Return all html filepaths."""
+        return self.files_to_lint
+
+    @property
+    def all_filepaths(self):
+        """Return all filepaths."""
+        return self.html_filepaths
+
+    def _check_html_tags_and_attributes(self, debug=False):
+        """This function checks the indentation of lines in HTML files."""
+
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting HTML tag and attribute check')
+            python_utils.PRINT('----------------------------------------')
+
+        html_files_to_lint = self.html_filepaths
+        stdout = sys.stdout
+
+        failed = False
+        summary_messages = []
+
+        with linter_utils.redirect_stdout(stdout):
+            for filepath in html_files_to_lint:
+                file_content = FILE_CACHE.read(filepath)
+                file_lines = FILE_CACHE.readlines(filepath)
+                parser = CustomHTMLParser(filepath, file_lines, debug)
+                parser.feed(file_content)
+
+                if len(parser.tag_stack) != 0:
+                    raise TagMismatchException('Error in file %s\n' % filepath)
+
+                if parser.failed:
+                    failed = True
+
+            if failed:
+                summary_message = (
+                    '%s   HTML tag and attribute check failed, fix the HTML '
+                    'files listed above.' % _MESSAGE_TYPE_FAILED)
+                python_utils.PRINT(summary_message)
+                summary_messages.append(summary_message)
+            else:
+                summary_message = '%s  HTML tag and attribute check passed' % (
+                    _MESSAGE_TYPE_SUCCESS)
+
+                python_utils.PRINT(summary_message)
+                summary_messages.append(summary_message)
+
+            python_utils.PRINT('')
+
+        return summary_messages
+
+    def perform_all_lint_checks(self):
+        """Perform all the lint checks and returns the messages returned by all
+        the checks.
+
+        Returns:
+            all_messages: str. All the messages returned by the lint checks.
+        """
+
+        if not self.all_filepaths:
+            python_utils.PRINT('')
+            python_utils.PRINT('There are no HTML files to lint.')
+            return []
+
+        # The html tags and attributes check has an additional
+        # debug mode which when enabled prints the tag_stack for each file.
+        html_tag_and_attribute_messages = (
+            self._check_html_tags_and_attributes())
```

Comment:
Just return the results of this function.

Reply:
Done.